### DISCUSSION TITLE
CartesianIndex hard to use

### DISCUSSION THREAD

**Discussion by trafficfan** [JuliaLang/julia#56738] [RFC: features for discussion]
```{julia}
# Orginal in Julia 0.6.x
(Tv[i,:],dr[i,:]) = findmax(u_temp+beta*Ev,1)
# in Julia 1.11.2
(Tv[i, :], inds) = findmax(u_temp + beta * Ev, dims=1)
dr[i, :]=[inds[1][2] inds[2][2]] 
```
using CartesianIndex makes such work hard to understand

**Comment by arnavk23** [comment]
Thanks for raising this — many users transitioning from older Julia versions hit the same friction.

In newer Julia versions (≥ 0.7), findmax(..., dims=...) returns a tuple:

   - First element: the max values

   -  Second: an array of CartesianIndex objects for the corresponding positions.

So the modern equivalent of:
```julia
(Tv[i,:], dr[i,:]) = findmax(u_temp + beta * Ev, 1)
```
would be something like:
```julia
vals, inds = findmax(u_temp + beta * Ev, dims=1)
Tv[i, :] = vals
dr[i, :] = getindex.(inds, 1)  # Get row indices from CartesianIndex
```
This change supports N-dimensional arrays more robustly, but yes, it's more verbose for simple 2D use-cases.
**Suggestions:**

  -  Maybe we could have a helper function like:
```julia
plain_indices(inds::Array{CartesianIndex}) = getindex.(inds, 1)
```
Or improved documentation/examples for common 2D array use.

**Comment by engineer-antonio** [comment]
The transition from v0.6 to v1.11 brought this change to make reduction operations consistent across all dimensions. CartesianIndex avoids ambiguities that occurred in the old Julia.

One way to make the code more readable and avoid using fixed indices like [1][2] is to convert the indices to tuples or use the ind2sub function (which is now done via Tuple(inds)):

Julia
(Tv[i, :], inds) = findmax(u_temp + beta * Ev, dims=1)
dr[i, :] = map(x -> x[2], inds)
